UnityProject/Assets/Scripts/Items/PDA/PDALogic.cs

```diff
@@ -111,48 +109,46 @@ private void Start()
 			CartridgeSlot = storage.GetIndexedItemSlot(1);
 
 			IDSlot.OnSlotContentsChangeServer.AddListener(OnIDSlotChanged);
-			//messengerSystem = GameObject.Find("MessengerManager").GetComponent<MessengerManager>();
-			//AddSelf();
-		}
-
-		public override void OnStartServer()
-		{
-			// TODO Instead, consider listening for client's OnPlayerSpawned and then request server to run stuff.
-			StartCoroutine(DelayInitialisation());
-		}
 
-		private IEnumerator DelayInitialisation()
-		{
-			yield return WaitFor.Seconds(TIME_BEFORE_UPLINK_ACTIVATION);
-			OnStartServerDelayed();
-		}
-
-		private void OnStartServerDelayed()
-		{
 			if (cartridgePrefab != null)
 			{
 				var cartridge = Spawn.ServerPrefab(cartridgePrefab).GameObject;
 				Inventory.ServerAdd(cartridge, CartridgeSlot);
 			}
+		}
+
+		#endregion Lifecycle
 
-			var owner = GetPlayerByParentInventory();
-			if (owner == null) return;
+		public void OnInventoryMoveServer(InventoryMove info)
+		{
+			// This is also triggered when the PDA spawns as part of the player's inventory populator
+			// and added to their inventory, when they spawn.
 
-			RegisteredPlayerName = owner.ExpensiveName();
-			TryInstallUplink(owner);
+			if (RegisteredPlayerName != default) return; // PDA already registered to someone
```

Comment:
I might be nitpicking, but instead of using ``default`` you could use ``string.IsNullOrEmpty()``

Reply:
I'm assigning the default value when resetting the string, so it made sense for me to check if it is still the default later.

Reply:
Yeah, and I think that works just fine. I suggested the other option because it is more obvious what you're asking. I will nonetheless merge now.